app/src/main/java/org/sagebionetworks/research/mpower/tracking/recycler_view/TriggersLoggingItemViewHolder.java

```diff
@@ -33,7 +32,7 @@ public TriggersLoggingItemViewHolder(final TriggersLoggingUIFormItemWidget itemV
 
     public void setContent(@NonNull SimpleTrackingItemConfig config) {
         this.config = config;
-        boolean logged = this.viewModel.isLogged(config);
+        boolean logged = this.viewModel.isLogged(config.getIdentifier());
```

Comment:
A RecyclerView ViewHolder is really only meant to be a class that gives access to views, it's not supposed to edit the data model directly. I'd suggest removing the ViewModel reference from this class and keeping it only in the Fragment.  The Fragment should listen to callbacks from the view holder for events like button presses.  This will also allow you to do row animations on the RecyclerView as well.